cli/tests/test_alias.rs

```diff
@@ -338,3 +343,41 @@ fn test_alias_in_repo_config() {
     aliases.l=["log", "-r@", "--no-graph", "-T\"user alias\\n\""]
     "###);
 }
+
+#[test]
+fn test_alias_help() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    test_env.add_config(r#"aliases.b = ["log", "-r", "@", "-T", "branches"]"#);
+    test_env.add_config(r#"aliases.s = ["status"]"#);
+    test_env.add_config(r#"aliases.f = ["git", "fetch"]"#); // test nested subcommand
```

Comment:
Might be useful to add an alias that's something like `b2 = ["b", "--no-graph"]` (or whatever you prefer) to test help for an alias defined in terms of another alias.

Reply:
this caught a bug, thank you! previously the help was actually non-deterministic based on the order we processed aliases in the hashmap. i changed it to a fixed-point algorithm.